enum.go

```diff
@@ -75,6 +80,28 @@ func (e Enum[M, V]) Parse(value V) *M {
 	return e.v2m[value]
 }
 
+// Match converts a raw value into a member like Parse. On the other hand,
+// this returns the matched member if V implements Matcher interface, .
+//
+// This is especially beneficial when the value type is struct, which
+// means that be able to implement a custom matcher.
+func (e Enum[M, V]) Match(value V) *M {
```

Comment:
Suggestion: we can make it a function instead of a method, and then we can put constraints on V. This will make the function type-safe and faster (by avoiding runtime type assertions). Something like this:

```suggestion
func Find[M iMember[V], V Matcher[V]](e Enum[M, V], value V) *M {
```

Reply:
Thanks for the good suggestions! I would like to adopt it. I will try it out once I have the interface name mentioned above.